snek/snek.py

```diff
@@ -130,6 +129,235 @@ def __add_error(self, message):
         self.errors.append((now, message))
         return (now, message)
 
+    def _find_files(self, base_path, extra_suffix=None):
+        """
+        Helper function to find files matching the allowed suffixes in the base_path.
+
+        Parameters
+        -----
+        base_path: str
+        extra_suffix: str
+
+        Notes
+        -----
+        If set and starts with '.', extra_suffix will be appended to the glob pattern.
+        If a file cannot be read, it will be skipped.
+
+        Returns
+        -------
+        list
+        """
+
+        glob_pattern = "**/*"
+
+        if extra_suffix and extra_suffix.startswith('.'):
+            glob_pattern = f"{glob_pattern}{extra_suffix}"
+
+        return [
+            p.resolve()
+            for p in Path(base_path).glob(glob_pattern)
+            if (
+                any([suffix in self.config.handlers for suffix in p.suffixes])
+                and p.is_file()
+                and os.access(p, os.R_OK)
+            )
+        ]
+
+    def __parse_frontmatter_from_filepath(self, filepath, metadata_only=False):
+        """
+        Gets frontmatter data from filepath. If metadata_only is set and True, returns only FM metadata.
+
+        Parameters
+        -----
+        filepath: path-like object
+
+        Notes
+        -----
+        Returns None if the frontmatter module used a handler not allowed in self.config.handlers.
+        
+
+        Returns
+        -------
+        dict
+        """
+
+        fm_data = frontmatter.load(filepath)
+        allowed_handlers = tuple(h['frontmatter_handler'] for h in self.config.handlers.values())
+        if not isinstance(fm_data.handler, allowed_handlers):
+            self.__add_error(f"{filepath} has invalid frontmatter data.")
+            return None
+
+        if metadata_only:
+            return fm_data.metadata
+        else:
+            return fm_data
+
+    def __parse_data_from_filepath(self, filepath):
```

Comment:
Since this method is only used in `__update_data_from_filepath()`, I would merge the two.